From 03e9e594242437f1c9b18ffa021d05d864c1e725 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Erik=20Bot=C3=B6?= <erik.boto@pelagicore.com>
Date: Fri, 5 Sep 2014 11:17:48 +0200
Subject: [PATCH] Disable gobject-introspection to enable build in Yocto.

Revert "all: Add gobject-introspection support"

This reverts commit 227efa02262a2e7e58ad40cfe823dd19cc9c03c9.

Conflicts:
	src/librygel-core/Makefile.am
	src/librygel-renderer/Makefile.am
	src/librygel-server/Makefile.am
---
 .gitignore                                |  2 --
 configure.ac                              |  2 --
 src/librygel-core/Makefile.am             | 21 ++-------------------
 src/librygel-renderer/Makefile.am         | 26 ++------------------------
 src/librygel-server/Makefile.am           | 27 +++------------------------
 src/librygel-server/rygel-serializer.vala |  2 +-
 6 files changed, 8 insertions(+), 72 deletions(-)

diff --git a/.gitignore b/.gitignore
index d8b6a29..8880b6b 100644
--- a/.gitignore
+++ b/.gitignore
@@ -20,8 +20,6 @@ stamp-*
 *.LOCAL.*
 *.BASE.*
 *.REMOTE.*
-*.gir
-*.typelib
 core
 *.log
 .intltool-merge-cache
diff --git a/configure.ac b/configure.ac
index 36b1434..236f3c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,8 +29,6 @@ LT_INIT([dlopen disable-static])
 
 AC_SYS_LARGEFILE
 
-GOBJECT_INTROSPECTION_REQUIRE([1.33.4])
-
 dnl Required versions of library packages
 dnl Not all of these are actually used, depending on the configure options.
 GLIB_REQUIRED=2.34.0
diff --git a/src/librygel-core/Makefile.am b/src/librygel-core/Makefile.am
index 8d113e2..4668b24 100644
--- a/src/librygel-core/Makefile.am
+++ b/src/librygel-core/Makefile.am
@@ -13,7 +13,6 @@ librygel_core_2_4_la_SOURCES = \
 
 librygel_core_2_4_la_VALAFLAGS = \
 	-H rygel-core.h -C --library=rygel-core-2.4 \
-	--gir=Rygel-2.4.gir \
 	--vapidir=$(srcdir) \
 	--pkg uuid \
 	--pkg posix \
@@ -47,21 +46,7 @@ DEPS_FILES = rygel-core-2.4.deps
 
 vapidir = $(VAPIDIR)
 vapi_DATA = $(VAPI_FILES) $(DEPS_FILES)
-rygel-core-2.4.vapi rygel-core.h Rygel-2.4.gir: librygel_core_2_4_la_vala.stamp
-
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS=
-INTROSPECTION_SCANNER_ARGS=
-INTROSPECTION_COMPILER_ARGS= -l librygel-core-2.4.so.2
-$(builddir)/RygelCore-2.4.gir: $(srcdir)/Rygel-2.4.gir
-	$(AM_V_GEN)$(SED) -e 's,Rygel\.,RygelCore.,g' -e 's,namespace name="Rygel",namespace name="RygelCore",g' $< >$@
-GIRS = RygelCore-2.4.gir
-
-girdir=$(datadir)/gir-1.0
-dist_gir_DATA = $(GIRS)
-
-typelibdir = $(libdir)/girepository-1.0
-nodist_typelib_DATA = $(GIRS:.gir=.typelib)
+rygel-core-2.4.vapi rygel-core.h: librygel_core_2_4_la_vala.stamp
 
 DISTCLEANFILES = RygelCore-2.4.typelib
 
@@ -72,6 +57,4 @@ EXTRA_DIST = \
 	$(VAPI_FILES) \
 	$(DEPS_FILES) \
 	uuid.vapi \
-	$(BUILT_SOURCES) \
-	RygelCore-2.4.gir \
-	Rygel-2.4.gir
+	$(BUILT_SOURCES)
diff --git a/src/librygel-renderer/Makefile.am b/src/librygel-renderer/Makefile.am
index 7150667..1b4b912 100644
--- a/src/librygel-renderer/Makefile.am
+++ b/src/librygel-renderer/Makefile.am
@@ -14,7 +14,6 @@ librygel_renderer_2_4_la_SOURCES = \
 
 librygel_renderer_2_4_la_VALAFLAGS = \
 	-H rygel-renderer.h --library=rygel-renderer-2.4 \
-	--gir=Rygel-2.4.gir \
 	$(LIBRYGEL_RENDERER_DEPS_VALAFLAGS) \
 	$(RYGEL_COMMON_LIBRYGEL_CORE_VALAFLAGS) \
 	$(RYGEL_COMMON_VALAFLAGS)
@@ -31,27 +30,6 @@ librygel_renderer_2_4_la_LDFLAGS = \
 vapidir = $(VAPIDIR)
 vapi_DATA = rygel-renderer-2.4.vapi rygel-renderer-2.4.deps
 
-rygel-renderer.h rygel-renderer-2.4.vapi Rygel-2.4.gir: librygel_renderer_2_4_la_vala.stamp
+rygel-renderer.h rygel-renderer-2.4.vapi: librygel_renderer_2_4_la_vala.stamp
 
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS=
-INTROSPECTION_SCANNER_ARGS=
-INTROSPECTION_COMPILER_ARGS= -l librygel-renderer-2.4.so.2 --includedir=$(builddir)/../librygel-core --includedir=$(srcdir)/../librygel-core
-$(builddir)/RygelRenderer-2.4.gir: $(srcdir)/Rygel-2.4.gir
-	$(AM_V_GEN)$(SED) -e 's,Rygel\.,RygelRenderer.,g' \
-		-e 's,namespace name="Rygel",namespace name="RygelRenderer",g' \
-		-e 's,RygelRenderer\.Plugin,RygelCore.Plugin,g' \
-		-e 's,RygelRenderer\.MediaDevice,RygelCore.MediaDevice,g' \
-		-e 's,<package name="rygel-renderer-2.4"/>,<include name="RygelCore" version="2.4"/>\n<package name="rygel-renderer-2.4"/>,' \
-		$< >$@
-GIRS = RygelRenderer-2.4.gir
-
-girdir=$(datadir)/gir-1.0
-dist_gir_DATA = $(GIRS)
-
-typelibdir = $(libdir)/girepository-1.0
-nodist_typelib_DATA = $(GIRS:.gir=.typelib)
-
-
-DISTCLEANFILES = RygelRenderer-2.4.typelib
-EXTRA_DIST = rygel-renderer.h rygel-renderer-2.4.vapi rygel-renderer-2.4.deps RygelRenderer-2.4.gir Rygel-2.4.gir
+EXTRA_DIST = rygel-renderer.h rygel-renderer-2.4.vapi rygel-renderer-2.4.deps
diff --git a/src/librygel-server/Makefile.am b/src/librygel-server/Makefile.am
index 75a0778..3ae773b 100644
--- a/src/librygel-server/Makefile.am
+++ b/src/librygel-server/Makefile.am
@@ -14,7 +14,6 @@ librygel_server_2_4_la_SOURCES = \
 
 librygel_server_2_4_la_VALAFLAGS = \
 	-H rygel-server.h --library=rygel-server-2.4 \
-	--gir=Rygel-2.4.gir \
 	$(LIBRYGEL_SERVER_DEPS_VALAFLAGS) \
 	$(RYGEL_COMMON_LIBRYGEL_CORE_VALAFLAGS) \
 	$(RYGEL_COMMON_VALAFLAGS)
@@ -35,26 +34,6 @@ librygel_server_2_4_la_LDFLAGS = \
 vapidir = $(VAPIDIR)
 vapi_DATA = rygel-server-2.4.vapi rygel-server-2.4.deps
 
-rygel-server.h rygel-server-2.4.vapi Rygel-2.4.gir: librygel_server_2_4_la_vala.stamp
-
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS=
-INTROSPECTION_SCANNER_ARGS=
-INTROSPECTION_COMPILER_ARGS= -l librygel-server-2.4.so.2 --includedir=$(builddir)/../librygel-core --includedir=$(srcdir)/../librygel-core
-$(builddir)/RygelServer-2.4.gir: $(srcdir)/Rygel-2.4.gir
-	$(AM_V_GEN)$(SED) -e 's,Rygel\.,RygelServer.,g' \
-		-e 's,namespace name="Rygel",namespace name="RygelServer",g' \
-		-e 's,RygelServer\.Plugin,RygelCore.Plugin,g' \
-		-e 's,RygelServer\.MediaDevice,RygelCore.MediaDevice,g' \
-		-e 's,<package name="rygel-server-2.4"/>,<include name="RygelCore" version="2.4"/>\n<package name="rygel-server-2.4"/>,' \
-		$< >$@
-GIRS = RygelServer-2.4.gir
-
-girdir=$(datadir)/gir-1.0
-dist_gir_DATA = $(GIRS)
-
-typelibdir = $(libdir)/girepository-1.0
-nodist_typelib_DATA = $(GIRS:.gir=.typelib)
-
-DISTCLEANFILES = RygelServer-2.4.typelib
-EXTRA_DIST = rygel-server.h rygel-server-2.4.vapi rygel-server-2.4.deps RygelServer-2.4.gir Rygel-2.4.gir
+rygel-server.h rygel-server-2.4.vapi: librygel_server_2_4_la_vala.stamp
+
+EXTRA_DIST = rygel-server.h rygel-server-2.4.vapi rygel-server-2.4.deps
diff --git a/src/librygel-server/rygel-serializer.vala b/src/librygel-server/rygel-serializer.vala
index 85507bc..0532aca 100644
--- a/src/librygel-server/rygel-serializer.vala
+++ b/src/librygel-server/rygel-serializer.vala
@@ -22,7 +22,7 @@
 
 using GUPnP;
 
-public enum Rygel.SerializerType {
+public enum SerializerType {
     /// Normal serialization of container/item using DIDL-Lite
     GENERIC_DIDL,
 
-- 
2.1.0

